cli/src/commands/codegen.rs

```diff
@@ -213,3 +249,23 @@ fn codegen(
     writeln!(output, "{code}")?;
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::commands::codegen::type_map_parser;
+
+    #[test]
+    fn parse_types() {
+        assert_eq!(type_map_parser("Foo"), None);
+        assert_eq!(type_map_parser("Foo=Bar"), Some(("Foo", "Bar", false)));
+        assert_eq!(
+            type_map_parser("Foo=Bar,recursive"),
+            Some(("Foo", "Bar", true))
+        );
+        assert_eq!(type_map_parser("Foo=Bar,a"), Some(("Foo", "Bar", false)));
+        assert_eq!(
+            type_map_parser("Foo=Bar,a,b,c,recursive"),
+            Some(("Foo", "Bar", true))
+        );
```

Comment:
Should we reject this format? 🤔 

Maybe we could also check that we expect this function to panic on these inputs since they are invalid

Reply:
Good point, let's reject it. To keep the possibility open to add other attributes (similar to `recursive`) in the future I implemented a simple match over the trailing attributes:
```rs
let mut recursive = false;
for r in split_rest {
    match r {
        "recursive" => {
            recursive = true;
        }
        _ => return None,
    }
}
```